### PR TITLE
Add func registry support for FilterQL statements. 

### PR DESCRIPTION
**Pull request by epsniff**
When using filter QL for evaluating different domains within the same binary.  We need to be able to isolate func resolution to specific subdomains.

### CODE REVIEW

File: rel/parse_filterql.go [hunk 1]
```diff
@@ -11,85 +11,106 @@ import (
...
+
+	buildVm bool
+	fs      *FilterStatement
+	l       *lex.Lexer
+	comment string
+	*filterTokenPager
+	firstToken lex.Token
+
+	funcs expr.FuncResolver
 }
```
Comment (epsniff) [inline rel/parse_filterql.go]: not sure if you like the builder pattern `NewFilterParser().Statement(filter).ParseFilter()`.  we can change it if you don't.

**Comment by araddon** [comment]
let me pull this and play with it

**Comment by epsniff** [comment]
@araddon cool, feel free to change it however you like :) .

**Comment by araddon** [comment]
played with it some more local, lgtm merging.  ty!

**Pull request merged**